02-algorithms/2-Searching/algos_searching_answers.txt

```diff
@@ -0,0 +1,22 @@
+1. Using linear search it would take 7 iterations to find G. 
+2. Using Binary search it would take 3 iterations to find G.
+3. (I’m confused why we’re doing this or how to demonstrate that I’m really doing this by hand but) the fib sequence is (I’m adding them in my head you’ll just have to believe me I guess (or maybe I’ll have screwed one up?) ??? 
+	1) 1 
+	2) 1 
+	3) 2 
+	4) 3
+	5) 5 
+	6) 8 
+	7) 13
+	8) 21
+	9) 34
+	10) 55
+	11) 89
+	12) 144
+
+(Note they don’t really ask for an answer but I’ve included a screen shot of my tests on the iterative and recursive functions - recursive seems to take longer, which based on my research has something to do with the language itself?)
+
+1. Given the unsorted list one would think we would need to use a linear search (despite it being a million long)… since the binary search wouldn’t have anything to measure the middle against on an unsorted list. I would think running it iteratively would be more effective most of the time. 
```

Comment:
If it were likely that I would have to search it twenty times or more, I would do a quicksort then binary search it. The theoretical performance of that method at twenty times totals about 19,931,967 operations versus 20 million for linear. It gets even crazier: at 120 usages, the linear search, of course, takes 120 million operations while the quicksort/binary search combo takes 19,933,960 ! One hundred more searches took our QB only two-thousand operations.

This is because quicksort is the big drain here. It grows waaaaay faster than linear search. However, you only need to do it once. So, once you get over the initial investment, binary search annihilates linear search at this magnitude.